### PR TITLE
Added Shift Form, Tables, Stories, and Tests

### PR DESCRIPTION
**Pull request by Aasish-Virjala**
In this PR, I added ShiftForm and ShiftTable components, as well as stories and tests. Screenshots attached below.

# Storybook

Screenshot 2023-08-27 at 6 03 08 PM
Screenshot 2023-08-27 at 6 03 19 PM
Screenshot 2023-08-27 at 6 03 33 PM
Screenshot 2023-08-27 at 6 03 42 PM

# Page:

Screenshot 2023-08-27 at 6 06 51 PM

### CODE REVIEW

File: frontend/src/tests/components/Shift/ShiftForm.test.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        );
+
+        expect(await screen.findByText(/Create/)).toBeInTheDocument();
+
+        expectedHeaders.forEach((headerText) => {
+            const header = screen.getByText(headerText);
+            expect(header).toBeInTheDocument();
+          });
+        expect(screen.getByTestId(`${testId}-cancel`)).toBeInTheDocument;
+        expect(screen.getByTestId(`${testId}-day`)).toBeInTheDocument;
```
Comment (pconrad) [inline frontend/src/tests/components/Shift/ShiftForm.test.js:38]: And on all of the rest as well.  That will get rid of one set of the eslint failures.

```suggestion
        expect(screen.getByTestId(`${testId}-day`)).toBeInTheDocument();
```

File: frontend/src/tests/pages/ShiftPage.test.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
-import { render, waitFor } from "@testing-library/react";
+import { render, waitFor, screen } from "@testing-library/react";
 import { QueryClient, QueryClientProvider } from "react-query";
 import { MemoryRouter } from "react-router-dom";
-import ShiftPage from "main/pages/ShiftPage";
+import ShiftPage from "main/pages/Shift/ShiftPage";
 import shiftFixtures from "fixtures/shiftFixtures";
-import { apiCurrentUserFixtures } from "fixtures/currentUserFixtures";
+import { apiCurrentUserFixtures, currentUserFixtures } from "fixtures/currentUserFixtures";
```
Comment (pconrad) [inline frontend/src/tests/pages/ShiftPage.test.js:6]: See feedback on slack

Comment (pconrad) [reply frontend/src/tests/pages/ShiftPage.test.js:6]: As well as the clear error message below.

File: frontend/src/tests/components/Shift/ShiftForm.test.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            </QueryClientProvider>
+        );
+
+        expect(await screen.findByText(/Create/)).toBeInTheDocument();
+
+        expectedHeaders.forEach((headerText) => {
+            const header = screen.getByText(headerText);
+            expect(header).toBeInTheDocument();
+          });
+        expect(screen.getByTestId(`${testId}-cancel`)).toBeInTheDocument;
```
Comment (pabucsb) [inline frontend/src/tests/components/Shift/ShiftForm.test.js:37]:
```suggestion
        expect(screen.getByTestId(`${testId}-cancel`)).toBeInTheDocument();
```

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**